simple/lint.go

```diff
@@ -656,12 +656,26 @@ func refersTo(pass *analysis.Pass, expr ast.Expr, ident types.Object) bool {
 }
 
 var checkLoopAppendQ = pattern.MustParse(`
+(Or
 	(RangeStmt
 		(Ident "_")
 		val@(Object _)
 		_
 		x
-		[(AssignStmt [lhs] "=" [(CallExpr (Builtin "append") [lhs val])])]) `)
+		[(AssignStmt [lhs] "=" [(CallExpr (Builtin "append") [lhs val])])])
+	(RangeStmt
+		key
+		nil
+		_
+		x
+		[(AssignStmt [lhs] "=" [(CallExpr (Builtin "append") [lhs (IndexExpr val@(Object _) key)])])])
```

Comment:
Shouldn't this be `(IndexExpr x key)` instead of `(IndexExpr val@(Object _) key)`?

Reply:
Thanks for review! I fixed at fd22dfa9520ac69c5615eb29b37beb2879eb55fd.